### ISSUE TITLE
Folders inside mnesia losing user 1001 and becoming root user and breaking RabbitMQ execution.

### ISSUE THREAD

**Issue by jluiz20** [bitnami/charts#16076]
### Name and Version

bitnami/rabbitmq 11.12.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

I could not find a way to reproduce it, it appears to happen randomly  

### Are you using any custom parameters or values?

yes, they are, but not any big change I added `fsGroupChangePolicy: "Always"` in hope to fix the problem, but it seems that for queue it still happening

```yaml
securePassword: false
loadDefinition:
 enabled: true
extraSecrets:
  load-definition:
    load_definition.json: |
      {
        "users": [
        {
          "name": "{{ .Values.auth.username }}",
          "password":  "{{ .Values.auth.password }}",
          "tags": "administrator"
        }
        ],
        "vhosts": [
          {
            "name": "/"
          }
        ],
        "permissions": [
          {
            "user": "{{ .Values.auth.username }}",
            "vhost": "/",
            "configure": ".*",
            "read": ".*",
            "write": ".*"
          }
        ],
        "policies": [
          {
            "name": "ha-all",
            "pattern": ".*",
            "vhost": "/",
            "definition": {
              "ha-mode": "all",
              "ha-sync-mode": "automatic"
            }
          }
        ]
      }

extraConfiguration: |-
  default_vhost = /
  default_permissions.configure = .*
  default_permissions.read = .*
  default_permissions.write = .*

tolerations: [
  {
    key: "dedicated",
    value: "rabbitmq",
    operator: "Equal",
    effect: "NoSchedule"
  }
]

podSecurityContext:
  runAsNonRoot: true
  fsGroupChangePolicy: "Always"

containerSecurityContext:
  capabilities:
    drop: ["NET_RAW"]

metrics:
  enabled: true

````

### What is the expected behavior?

folders to keep with user `1001`

### What do you see instead?

I noticed the problem happening in two different places, one in the `mnesia` folder and also sometimes in a specific queue.

When I install the chart, this is the users and permissions that I see on the mnseia folder (`bitnami/rabbitmq/mnesia`) 

![image](https://user-images.githubusercontent.com/9250618/232251924-9ec6318e-6ec2-4f96-a625-3d92b0348d0c.png)

you can notice that all the `rabbitmq@...` folders have user 1001 as expected. But after some time (days, months), for some reason, RabbitMQ stops working and when I list the users again, the user has changed from `1001` to `root` and the RabbitMQ has no access anymore, as seen in the image below.

![image](https://user-images.githubusercontent.com/9250618/232262047-c900720d-57ec-474e-bfa4-f5091b1f269c.png)

After adding the flag `podSecurityContext.fsGroupChangePolicy: "Always"` I didn't see it again, not sure if it really fixed or not, as I don't have a way to reproduce it.

Although the problem with the entire mnesia folder didn't happen again, I see now that some specific queue user changed for some unknown reason.

the image below shows the list of queues, you can notice that the queue `ENIK9V2RN0WTX72CTIU7ISYZY` lost the user `1001` and now it is running on `root`. with that, I cannot do anything, like delete it, connect, or anything.

![image](https://user-images.githubusercontent.com/9250618/232252211-f782790e-ac82-403c-9aa2-838c07d21357.png)

For example, when I try to delete it via the dashboard, I get the following error

<img width="785" alt="image" src="https://user-images.githubusercontent.com/9250618/232252359-c20f9809-e065-49c2-ba7a-c20001a22ed6.png">

the only way that I found that "fix" it is to restart RabbitMQ. But the problem happens again after a few days/months.

I tried many things, but I don't have a clue about what can be happening. I also don't know if it is something related to the chart or image itself (or something external)

### Additional information

we are running chart `version: 11.12.0` (`appVersion: 3.11.11` ) in Kubernetes `v1.24.11`. 

Our queues and exchange definitions are created by the application, which uses Spring AMPQ for java based and amqplib for node services. 

We are planning to use this chart to replace the current one that we use (that is deprecated). The old chart (which is not Bitnami) runs with RabbitMQ 3.8.5 and we never had this problem with that chart.

I did a test with the Bitnami chart (with the RabbitMQ version 3.8.5) and I saw the same problem happening. Not sure if the cause can be in the Bitnami RabbitMQ image.

Honestly, I don't have any clue. I tried everything that I know (and don't know) and this still happens. Worse is that I don't have a reliable way to replicate and check if it some changes make difference.

Any help is really appreciated.

**Comment by jluiz20** [comment]
Just to complement, this is the log file before the specific queue loses user `1001` and becomes `root`. 

It starts with

```
2023-04-13 11:07:33.213830+00:00 [error] <0.32355.20> ** Generic server <0.32355.20> terminating
2023-04-13 11:07:33.213830+00:00 [error] <0.32355.20> ** Last message in was pre_hibernate
2023-04-13 11:07:33.213830+00:00 [error] <0.32355.20> ** When Server state == {q,
```

and then we just get timeout and cannot do anything anymore

[rabbitmq-ha-0-ci-dev2-with queue without access copy.log](https://github.com/bitnami/charts/files/11242964/rabbitmq-ha-0-ci-dev2-with.queue.without.access.copy.log)

**Comment by carrodher** [comment]
It seems a very specific use case difficult to reproduce on our side and very tied to your scenario.

For information regarding the application itself, customization of the content within the application, or questions about the use of the technology or infrastructure; we highly recommend checking forums and user guides made available by the project behind the application or the technology.

That said, we will keep this ticket open until the stale bot closes it just in case someone from the community adds some valuable info.

**Comment by jluiz20** [comment]
Yeah, it is very difficult to reproduce. I was not able to find a way to reproduce it consistently. 

Based on the time that I spent debugging this, I believe that is something in the way the Bitnami container image is built, as I even did a test using the version of this chart that used version 3.8.5 of RabbiMQ (the version that we currently use in PROD, but we use the official image now) and we still had this problem.

I also thought it could be k8s version or something as I started we were using 1.23, but we updated to 1.24 and the problem still happens.

I am not sure what could be causing a folder to change user permission.

Anyway, thank you for the reply! I will probably stick with the original container image for a while.
